### DISCUSSION TITLE
focus-visible: vs focus: in `<input>`

### DISCUSSION THREAD

**Discussion by severinlandolt** [tailwindlabs/tailwindcss#13071] [Help] [answered]
Hi all!

I am trying to replicate the following behaviour: https://vercel.com/design/input
![CleanShot 2024-03-02 at 20 07 44@2x](https://github.com/tailwindlabs/tailwindcss/assets/41927988/25dad208-21a7-4f78-aeef-6e7fe8cac3b3)

If you click on the input, its getting a grey box shadow. If tabbed (keyborad), it's getting a blue outline. I thought this would be easily replicated using focus and focus-visible, but it somehow does not work on the input. 

Heres how far I got. https://play.tailwindcss.com/VdqGkaNvCT
Works with button, but with input its always focused, regardless if clicked or by keyboard.

**Comment by wongjn** [comment, marked as answer]
You may need to use JavaScript to differentiate between the two different focusing methods. At least, that's how the example that you linked does it.
